Lib/html/parser.py

```diff
@@ -57,6 +58,26 @@
 # </ and the tag name, so maybe this should be fixed
 endtagfind = re.compile(r'</\s*([a-zA-Z][-.a-zA-Z0-9:_]*)\s*>')
 
+# Character reference processing logic specific to attribute values
+# See: https://html.spec.whatwg.org/multipage/parsing.html#named-character-reference-state
+attr_charref = re.compile(r'&(#[0-9]+|#[xX][0-9a-fA-F]+|[a-zA-Z][a-zA-Z0-9]*)[;=]?')
+
+def _replace_attr_charref(match):
+    ref = match.group(0)
+    # Numeric / hex char refs must always be unescaped
+    if ref.startswith('&#'):
+        return unescape(ref)
+    # Named character / entity references must only be unescaped
+    # if they are an exact match, and they are not followed by an equals sign
+    terminates_with_equals = ref.endswith('=')
+    exact_match = ref.lstrip('&').rstrip('=') in html5_entities
+    if exact_match and not terminates_with_equals:
+        return unescape(ref)
+    # Otherwise do not unescape
+    return ref
```

Comment:
If `terminates_with_equals` is false, `rstrip('=')` has not effect.

The code can even be rewritten as
```py
    if not ref.endswith('=') and ref[1:] in html5_entities:
        return unescape(ref)
    return ref
```
We can even use captured groups to check for `=` and strip `&`.

Reply:
Thanks, updated the code with your suggestion